module_1/src/MovieListPage/MovieListPage.js

```diff
@@ -1,106 +1,108 @@
-    const backButton = <FontAwesomeIcon icon={faArrowLeft} />;
+    const defaultGenre = mainGenres[0];
+    const [activeGenre, setActiveGenre] = useState(searchParams.get('genre')?.toUpperCase() || defaultGenre);
+    const navigate  = useNavigate();
+    const [searchQuery, setSearchQuery] = useState(searchParams.get('query') || '');
     const moviesUrl = `http://localhost:4000/movies`;
+    const backButton = <FontAwesomeIcon icon={faArrowLeft} />;
 
-    useEffect(()=> {
-        fetch(`${moviesUrl}?search=${searchQuery}&searchBy=title`)
-        .then((response) => response.json())
-        .then((json)=> {
-            setMovies(json.data);
-            setTotalMovieAmount(json.data.length);
-        });
-    }, [searchQuery]);
 
     useEffect(()=> {
-        fetch(`${moviesUrl}?sortBy=${sortCriterion}&sortOrder=asc`)
-        .then((response) => response.json())
-        .then ((json)=> {
-            setMovies(json.data);
-            setTotalMovieAmount(json.data.length)
-        });
-    }, [sortCriterion]);
+        debugger;
+        fetch(getRequestUrl())
+            .then((response) => response.json())
+            .then((json)=> {
+                setMovies(json.data);
+                setTotalMovieAmount(json.data.length);
+            });
+            updateSearchParams();
+    }, [searchQuery, sortCriterion, activeGenre]);
 
-    function getMovieDetails(id) {
-        fetch(`${moviesUrl}/${id}`)
-        .then((response)=> response.json())
-        .then((json)=> {
-            defineMovieDetails(json);
-        })
-    };
+    function updateSearchParams(){
+        if (searchQuery) {
+            searchParams.set('query', searchQuery);
+        } else {
+            searchParams.delete('query');
+        }
 
-    function defineMovieDetails(json) {
-        let {title, genres, poster_path, release_date, vote_average, runtime, overview  } = json;
-            release_date = release_date.slice(0,4);
-            setMovieDetails({poster_path, title, genres, release_date, vote_average, runtime, overview});
+        searchParams.set('sortBy', sortCriterion.toLowerCase().replace(' ', '_'));
+
+        if (activeGenre === defaultGenre) {
+            searchParams.delete('genre');
+        } else {
+            searchParams.set('genre', activeGenre.toLowerCase());
+        }
+        setSearchParams(searchParams);
+    }
+
+    function getMovieDetails(id) {
+        navigate(`/${id}`);
     };
 
-    function filterMovies (e) {
-        e.preventDefault();
-        const selectedGenre = e.currentTarget.innerHTML;
-        setIsGenreActive(selectedGenre);
+    function getRequestUrl(selectedGenre) {
         let url = moviesUrl;
-        if (selectedGenre !== 'ALL') {
-            url += `?filter=${selectedGenre}`;
+        let hasParameter = false;
+        const genre = selectedGenre || activeGenre;
+        if (genre !== 'ALL') {
+            url += `?filter=${genre.toLowerCase()}`;
```

Comment:
You may find useful this API to work with search queries - https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams